src/component/contents/DraftEditorContents.react.js

```diff
@@ -192,26 +194,88 @@ class DraftEditorContents extends React.Component {
         <Component {...componentProps} />,
       );
 
+      // wrapperTemplate is the internal DraftJS wrapping template to wrap various blocks.
+      // customWrapperTemplate is a user provided component 
+      // and will also wrap blocks on top of wrapping by wrapperTemplate.
       processedBlocks.push({
         block: child,
-        wrapperTemplate,
         key,
         offsetKey,
+        wrapperTemplate,
+        customWrapperTemplate,
       });
 
-      if (wrapperTemplate) {
+      if (wrapperTemplate || customWrapperTemplate) {
         currentDepth = block.getDepth();
       } else {
         currentDepth = null;
       }
       lastWrapperTemplate = wrapperTemplate;
+      lastCustomWrapperTemplate = customWrapperTemplate;
     }
 
-    // Group contiguous runs of blocks that have the same wrapperTemplate
+    // Group contiguous runs of blocks
     const outputBlocks = [];
-    for (let ii = 0; ii < processedBlocks.length; ) {
+    for (let ii = 0; ii < processedBlocks.length;) {
       const info = processedBlocks[ii];
-      if (info.wrapperTemplate) {
+
+      // Group by customWrapperTemplate (if exists)
+      // Note: If we ever want to remove customWrapperTemplates, remove this if block
+      // customWrapperTemplate will always wrap the internal wrapperTemplate
+      if (info.customWrapperTemplate) {
+        // Group by custom template
+        const blocksGroupedByCustomWrapper = [];
+        do {
+          blocksGroupedByCustomWrapper.push(processedBlocks[ii]);
+          ii++;
+        } while (
+          ii < processedBlocks.length &&
+          processedBlocks[ii].customWrapperTemplate == info.customWrapperTemplate
+        );
+
+        // After grouping by customWrapperTemplate, 
+        // group those blocks by wrapperTemplate (internal wrapper)
+        const blocksGroupedByInternalWrapper = [];
+        for (let j = 0; j < blocksGroupedByCustomWrapper.length;) {
+          const accum = [];
```

Comment:
Maybe `blocksInCustomWrapper`? Alternatively, you could name it just `blocks` so it's clearer that this serves the same purpose as the `blocks` array declared first in the do/while for just `internalWrapperTemplate`.

Reply:
I like blocksInCustomWrapper. I'd rather not name it `blocks` as variable shadowing can lead to some frustrating situations in the future